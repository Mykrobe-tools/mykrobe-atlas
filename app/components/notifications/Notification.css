.wrap {
  background: var(--color-app-blue);
  border-bottom: 1px solid white;
  color: white;
  display: block;
  padding: 10px;
}

.message {
  background: var(--color-app-blue);
  color: white;
}

.error {
  background: var(--color-app-red);
  color: white;
}

.success {
  background: var(--color-app-green);
  color: white;
}

.content {
  composes: fontSecondary from "css/typography.css";
  composes: fontSecondarySizeNormal from "css/typography.css";
}
