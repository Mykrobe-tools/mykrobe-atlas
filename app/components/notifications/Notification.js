/* @flow */

import * as React from 'react';
import PropTypes from 'prop-types';

import styles from './Notification.css';

class Notification extends React.Component {
  onClick(event: Event) {
    const { onClick, id } = this.props;
    event.preventDefault();
    onClick(id);
  }

  render() {
    const { category, content } = this.props;
    let icon;
    switch (category) {
      case 'success':
        icon = 'fa fa-check-circle';
        break;
      case 'message':
        icon = 'fa fa-info-circle';
        break;
      case 'error':
        icon = 'fa fa-exclamation-circle';
        break;
    }
    return (
      <a
        className={styles[category]}
        onClick={event => this.onClick(event)}
        data-tid="notification"
      >
        <p className={styles.content}>
          {icon && (
            <span className={styles.icon}>
              <i className={icon} />
            </span>
          )}
          {content}
        </p>
        <span className={styles.close}>
          <i className="fa fa-times-circle" />
        </span>
      </a>
    );
  }
}

Notification.propTypes = {
  category: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
};

export default Notification;
