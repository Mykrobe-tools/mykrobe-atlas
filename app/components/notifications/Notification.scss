@import '~styles/colors.global';

.container {
	padding: 0.75rem 0.5rem;
	display: flex;
	flex-direction: column;
	line-height: 1.2;
	pointer-events: auto;
	position: relative;
	overflow: hidden;
}

.containerSeparate {
	composes: container;
	background-color: white;
	border-radius: 3/16 * 1rem;
}

.containerSeparate + .containerSeparate {
	margin-top: 0.25rem;
}

.containerJoined {
	composes: container;
	composes: global(border-bottom);
	background-color: white;
}

.iconBase {
	padding-right: 0.5rem;
	font-size: 16/16 * 1rem;
}

.messageIcon {
	composes: iconBase;
	color: $info;
}

.errorIcon {
	composes: iconBase;
	color: $danger;
}

.successIcon {
	composes: iconBase;
	color: $success;
}

.contentAndButtons {
	display: flex;
	flex-direction: row;
}

.content {
	overflow: hidden;
}

.contentConstrained {
	white-space: nowrap;
	overflow: hidden;
	text-overflow: ellipsis;
}

.buttons {
	display: flex;
	flex-direction: row;
	margin-left: auto;
}

.button {
	padding-left: 0.5rem;
	color: $color-grey-mid;
	font-size: 16/16 * 1rem;
}

.actions {
	display: flex;
	flex-direction: row;
	margin-left: auto;
}

.progress {
	position: absolute;
	left: 0;
	top: 0;
	height: 0.25rem;
	background-color: $color-tint-primary;
}
