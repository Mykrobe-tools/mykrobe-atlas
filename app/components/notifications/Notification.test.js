import * as React from 'react';
import renderer from 'react-test-renderer';
import Notification from './Notification';

describe('Notification component snapshot', () => {
  const id = 1;
  const category = 'success';
  const content = 'Lorem Ipsum';
  const onClick = jest.fn();

  it('renders correctly', () => {
    const component = renderer.create(
      <Notification
        id={id}
        category={category}
        content={content}
        onClick={onClick}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('runs click event', () => {
    const component = renderer.create(
      <Notification
        id={id}
        category={category}
        content={content}
        onClick={onClick}
      />
    );
    let tree = component.toJSON();
    tree.props.onClick({ preventDefault: jest.fn() });
    expect(onClick).toBeCalledWith(1);
  });
});
