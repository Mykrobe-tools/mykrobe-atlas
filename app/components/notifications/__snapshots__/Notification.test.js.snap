// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification component snapshot renders correctly 1`] = `
<a
  className="success"
  data-tid="notification"
  onClick={[Function]}
>
  <p
    className="content"
  >
    <span
      className="icon"
    >
      <i
        className="fa fa-check-circle"
      />
    </span>
    Lorem Ipsum
  </p>
  <span
    className="close"
  >
    <i
      className="fa fa-times-circle"
    />
  </span>
</a>
`;
